### DISCUSSION TITLE
Trying to compile for ARM v7 on M1 runners

### DISCUSSION THREAD

**Discussion by jfalcou** [actions/virtual-environments#10665] [General questions]
Hi,
I am the comaintainer of [EVE](https://github.com/jfalcou/eve), a C++ SIMD wrapper.
We wanted to migrate from our self hosted, cross compiling and qemu based ARM support to the new M1 runners.
It works perfectly for AARCH64 but I can't find a proper setup to compile for NEON only.

As EVE needs to be tested on all thing SIMD to ensure its quality, I was wondering if anyone had a way to do so.
Using the usual --target etc fails as it looks like the clang++ from homebrew just dont want to support it and looking at [the log](https://github.com/jfalcou/eve/actions/runs/10972646762/job/30469157445), looks like -arch arm64 is auto-added anyway.

Any hints appreciated.

**Comment by erik-bershel** [comment]
Hey @jfalcou!

I am not so familiar with this kind of edge case. I am not sure that `M1` fully support old `ARM v7` arch. But it should be available via cross-compilation at least, I suppose. So, I have at the moment just couple of silly ideas:

- to try to override `CMake` System Detection in toolchain file (dangerous area):

```cmake
set(CMAKE_SYSTEM_PROCESSOR armv7-a)
set(CMAKE_SYSTEM_NAME Linux)
``` 

- to explicitly target on NEON with something like that (not sure about `target`):

```bash
set(CMAKE_CXX_FLAGS "-mfpu=neon -march=armv7-a -mfloat-abi=hard -target arm-linux-gnueabihf")
```

But two questions remain open - does M1 support cross-compilation for the old architecture and does LLVM contain libraries for such cross-compilation. In the second case, you will have to rebuild LLVM instead of the one installed from brew.
